modules/generic_worker/templates/run-generic-worker.sh.erb

```diff
@@ -14,6 +14,39 @@ rm -f tasks-resolved-count.txt
 # wrapper script...
 /usr/local/bin/generic-worker "$@" 2>&1 | logger -t generic-worker -s
 
+exitstatus=( ${PIPESTATUS[0]} )
```

Comment:
I wasn't familiar with `PIPESTATUS` - nice job!

Nit: I suspect it doesn't require the outer brackets, e.g. maybe this is sufficient:
```
exitstatus=${PIPESTATUS[0]}
```

Reply:
exitstatus=${PIPESTATUS[0]} sisn't work. you need to put ${PIPESTATUS[0]} between ()